### PR TITLE
Chore/upstream merge

### PR DESCRIPTION
**Pull request by jordy25519**
Lot's of block synchronization fixes and internal refactors from upstream

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -0,0 +1,528 @@
+= Substrate
```
Comment (cowboy-bebug) [inline README.adoc]: Do we want to remove this file?

**Review by cowboy-bebug — CHANGES_REQUESTED** [review]
Looks good! I think we just need to remove `README.adoc` and edit/delete `CONTRIBUTING.adoc` as well (it keeps popping up when making a PR)

File: node/primitives/src/lib.rs [hunk 2]
```diff
@@ -28,7 +28,7 @@ use sr_primitives::{
 pub type BlockNumber = u32;
 
 /// Alias to 512-bit hash when used in the context of a transaction signature on the chain.
-pub type Signature = MultiSignature;
+pub type Signature = MsultiSignature;
```
Comment (aliXsed) [inline node/primitives/src/lib.rs]: Is this an expected change?

File: srml/executive/tests/doughnut_integration.rs [hunk 3]
```diff
@@ -150,7 +150,7 @@ impl transaction_payment::Trait for Runtime {
 	type FeeMultiplierUpdate = ();
 }
 
-#[allow(deprecated)]
+#[allow(deprecated)] // Allow ValidateUnsigned
```
Comment (aliXsed) [inline srml/executive/tests/doughnut_integration.rs]: The comment is not expressive to me.

File: node/primitives/src/lib.rs (hunk 2 above)
Comment (aliXsed) [reply node/primitives/src/lib.rs]: Okay! I saw it was fixed in the following commit.

**Review by aliXsed — APPROVED** [review]

**Review by cowboy-bebug — APPROVED** [review]
Yep looks good!

**Comment by cowboy-bebug** [comment]
We might need to remove signing commits for this one 😄

**Pull request merged**